src/mds/MDSRank.cc

```diff
@@ -1430,7 +1429,7 @@ void MDSRank::send_message_mds(const ref_t<Message>& m, mds_rank_t mds)
   auto addrs = mdsmap->get_addrs(mds);
   if (mds != whoami && peer_mdsmap_epoch[mds] < mdsmap->get_epoch()) {
     auto _m = make_message<MMDSMap>(monc->get_fsid(), *mdsmap,
-				    std::string(mdsmap->get_fs_name()));
+                                    std::string());
```

Comment:
You can remove this argument from the MMDSMap constructor. It can internally always save an empty string in the encoding.

Reply:
Done.